app.rb

```diff
@@ -0,0 +1,76 @@
+require_relative './book'
+require_relative './student'
+require_relative './teacher'
+require_relative './rental'
+
+def list_all_books(books)
+  if books.empty?
+    puts 'There are no books'
+  else
+    books.each { |book| puts "Title: '#{book.title}' Author: '#{book.author}'" }
+  end
+end
+
+def list_all_people(people)
+  if people.empty?
+    puts 'There are no people listed'
+  else
+    people.each { |person| puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
+  end
+end
+
+def create_person(person)
+  puts 'do you want to create a student(1) or a teacher (2):'
+  selected_person = gets.chomp.to_i
+  print 'age:'
+  age = gets.chomp.to_i
+  print 'name:'
+  name = gets.chomp.to_s
+  case selected_person
+  when 1
+    print 'Has parent permission? [Y/N]: '
+    provided_permission = gets.chomp.capitalize
+    student_permission = true if provided_permission == 'Y'
+    student_permission = false if provided_permission == 'N'
+    person.push(Student.new(nil, age, name, parent_permission: student_permission))
+  when 2
+    print 'Specialization: '
+    specialization = gets.chomp.to_s
+    person.push(Teacher.new(age, name, specialization))
+  end
+  puts 'Person created successfully.'
+end
+
+def create_book(books)
+  print 'title:'
+  title = gets.chomp
+  print 'author:'
+  author = gets.chomp
+  books.push(Book.new(title, author))
+  puts 'the book is created successfully'
+end
+
+def create_rental(books, people, rentals)
+  puts 'Select a book from the following list by number'
+  books.map.with_index { |book, index| puts "#{index}) Title: '#{book.title}', Author: #{book.author}" }
+  selected_book = gets.chomp.to_i
+
+  puts 'Select a person from the following list by number (Not ID): '
+  people.map.with_index do |person, index|
+    puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+  end
+
+  selected_person = gets.chomp.to_i
+  print 'date:'
+  provided_date = gets.chomp
+  rentals.push(Rental.new(provided_date, people[selected_person], books[selected_book]))
```

Comment:
- [ ] The app crashes and quits when I try to list rentals for a person. This is because on this line your arrangement of arguments is different from what you defined in the Rental class constructor, and so the app is looking for an id in the book object, instead of the person object.

  In the Rental class your order of parameters is this: `(date, book, person)`, but here you are passing the arguments in the order `(date, person, book)`. You can fix this by do this instead

  ```
  rentals.push(Rental.new(provided_date, books[selected_book], people[selected_person]))
  ```
  ![ren](https://user-images.githubusercontent.com/81468675/189325539-cfd03eb9-40dc-4188-a697-03525442de1c.jpg)